src/ui/atoms/textField/TextField.tsx

```diff
@@ -20,48 +20,28 @@ export type TextFieldProps = InputBaseProps & {
    * If true, the TextField will take 100% of its parent's size.
    */
   readonly fullWidth?: boolean
-  /**
-   * If true, the TextField allows multiline text as a text area.
-   */
-  readonly multiline?: boolean
-  /**
-   * If multiline, defines the number of lines of the text area.
-   */
-  readonly numberOfLines?: number
-  /**
-   * If true and multiline, the user can't resize the text area.
-   */
-  readonly disableAreaResize?: boolean
 }
 
 export const TextField: React.FC<TextFieldProps> = ({
   size = 'medium',
   helperText,
-  fullWidth = false,
-  multiline = false,
-  numberOfLines,
-  disableAreaResize = false,
   ...props
 }: TextFieldProps): JSX.Element => {
+  const { multiline, disableAreaResize, fullWidth, hasError }: InputBaseProps = props
   const containerClass = classNames(
     `okp4-text-field-main ${multiline && !disableAreaResize ? 'auto' : size}`,
     {
-      'full-width': !multiline && fullWidth
+      'full-width': fullWidth
```

Comment:
I don't know if it's possible, but I will expect this className to be carried directly by the `InputBase` base style. 
Tell me if I'm wrong 😌

Reply:
I think we can't because the `fullWidth` props is used by a container that contains the `InputBase` 

```tsx
const containerClass = classNames(
    `okp4-text-field-main ${multiline && !disableAreaResize ? 'auto' : size}`,
    {
      'full-width': fullWidth
    }
  )

  return (
    <div className={containerClass}>
      <InputBase {...props} />
      {helperText && (
        <Typography
          as="div"
          color={hasError ? 'error' : 'info'}
          fontSize="x-small"
          fontWeight="bold"
          noWrap
        >
          {helperText}
        </Typography>
      )}
    </div>
  )
  ```

Reply:
What do you think @lolottetheclash ?

Reply:
At first sight I would say that this classname must be managed here in order to wrap the input and the helperText. Otherwise, it would be necessary to refactor the whole thing in order to lower the management of the classname by one level.

Reply:
OK I see the point, but, IMHO, the way of dealing with the `fullWidth` property is not correct.
Doing like this forces all descendants to declare their own way of handling this property, so there Is no genericity at all.
I think this should be part of a little refactoring that we should address soon if you are OK 😌

Reply:
ok for me